experimental/dds/tree2/src/test/feature-libraries/contextuallyTyped.spec.ts

```diff
@@ -81,10 +97,115 @@ describe("ContextuallyTyped", () => {
 		const numberSequence = SchemaBuilder.fieldSequence(numberSchema);
 		const primaryObject = builder.object("numbers", { local: { [EmptyKey]: numberSequence } });
 		const schema = builder.intoDocumentSchema(numberSequence);
-		const mapTree = applyTypesFromContext(schema, new Set([primaryObject.name]), []);
+		const mapTree = applyTypesFromContext(
+			{ schema, getFieldGenerator: defaultGetFieldGenerator },
+			new Set([primaryObject.name]),
+			[],
+		);
 		const expected: MapTree = { fields: new Map(), type: primaryObject.name, value: undefined };
 		assert.deepEqual(mapTree, expected);
 	});
 
+	describe("cursorFromContextualData adds globalFieldKey", () => {
+		it("for empty contextual data.", () => {
+			const builder = new SchemaBuilder("Identifier Domain");
+			const identifierSchema = builder.primitive("identifier", ValueSchema.String);
+			const identifierFieldSchema = builder.globalField(
+				"identifier",
+				SchemaBuilder.fieldValue(identifierSchema),
+			);
+			const nodeSchema = builder.objectRecursive("node", {
+				local: {
+					child: SchemaBuilder.fieldRecursive(FieldKinds.optional, () => nodeSchema),
+				},
```

Comment:
you should be able to make your simple unit test here simpler. For example, you should be able to remove support for this optional child node, making the schema not recursive and simpler.

Reply:
removed local field, and made schema non-recursive